api/src/main/java/org/openmrs/module/fhirExtension/export/impl/PatientExport.java

```diff
@@ -16,17 +17,25 @@ public class PatientExport implements Exporter {
 	
 	private FhirPatientService fhirPatientService;
 	
+	private AnonymiseHandler anonymiseHandler;
+	
 	@Autowired
-	public PatientExport(FhirPatientService fhirPatientService) {
+	public PatientExport(FhirPatientService fhirPatientService, AnonymiseHandler anonymiseHandler) {
 		this.fhirPatientService = fhirPatientService;
+		this.anonymiseHandler = anonymiseHandler;
 	}
 	
 	@Override
-	public List<IBaseResource> export(String startDate, String endDate) {
+	public List<IBaseResource> export(String startDate, String endDate, boolean isAnonymise) {
 		DateRangeParam lastUpdated = getLastUpdated(startDate, endDate);
 		PatientSearchParams patientSearchParams = new PatientSearchParams(null, null, null, null, null, null, null, null,
 		        null, null, null, null, null, lastUpdated, null, null);
 		IBundleProvider iBundleProvider = fhirPatientService.searchForPatients(patientSearchParams);
-		return iBundleProvider.getAllResources();
+		return isAnonymise ? anonymise(iBundleProvider.getAllResources()) : iBundleProvider.getAllResources();
+	}
+	
+	private List<IBaseResource> anonymise(List<IBaseResource> iBaseResources) {
```

Comment:
All implementations of Exporter have this redundant method 
List<IBaseResource> anonymise(List<IBaseResource> iBaseResources)
Add additional parameter i.e. resource type and move it to default method in the interface